doc/Coding Guidelines.md

```diff
@@ -0,0 +1,107 @@
+# Vue Storefront Coding Guidelines - DRAFT
+
+Hello Extension / Theme / Core developer! This is just a short guide to align the code style among the core modules and all the extensions and thems.
+
+## General guidelines
+In general we're using kind of mix of: [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript) and [JavaScript Standard Style](https://github.com/feross/standard/blob/master/RULES.md#javascript-standard-style). There are a eslint and vue-eslint configured to enforce the style. Please do use the same [configs in Your modules](https://github.com/DivanteLtd/vue-storefront/blob/master/.eslintrc.js). 
+
+## ES2015
+We do preffer to use ES2015 constructions in the code - and there is a babel compiler set to allow us to do so. So use the ES modules instead of `require` and feel free to use other fancy structures.
+
+## Components
+Components should be created as [Single File Vue components](https://vuejs.org/v2/guide/single-file-components.html) and names using `CamelCase.vue`.
+All the component properties should be named using `camelCase` as well.
+All the component data fields should be `camelCased`.
+
+Theme catalog structure is well described in here: [Working with themes](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/themes/Working%20with%20themes.md)
+
+## Extensions
+For the reference of how extension can be structured please take a look at [template](https://github.com/DivanteLtd/vue-storefront/tree/develop/src/extensions/template).
+Check the [Working with extensions](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/extensions/Working%20with%20extensions.md) for more info.
+
+It's something like this:
+
+```
+|____store.js
+|____index.js
+|____components
+| |____ComponentExample.vue
+|____router.js
+|____package.json
+|____pages
+| |____PageExample.vue
+```
+
+## Events
+We're using [Vue events](https://vuejs.org/v2/guide/events.html) for async. operations handling and exposing some extension points in the code.
+
+The naming convention for the new events to be introduced is:
+`module-[before|after]-operation` - for example the event fired after order has been placed is named: `order-after-placed` and the event fired after product is loaded: `product-after-load`. 
+
+## Vuex
+We're using Vuex for all kind of state/date related operations. Vuex stores are modularized and Your extensions can have additional Vuex modules as well by just extending it's own [`store.js`](https://github.com/DivanteLtd/vue-storefront/tree/master/src/extensions/custom_extension).
+If You're adding some data operations to Core - check if t shouldn't be another module for that, then create the store separating the `actions`, `mutations` and `getters` to separate files:
+
+```
+.
+|____order
+| |____mutations.js
+| |____actions.js
+| |____index.js
+|____attribute
+| |____mutations.js
+| |____actions.js
+| |____getters.js
+| |____index.js
+| |____helpers.js
+|____homepage
+| |____index.js
+...
+```
```

Comment:
I think we should list all store modules and show the general structure separately.

Something like this:

Store modules:
```
- attribute
- homepage
- order
```
Single module structure:
```
- actions.js
- getters.js
- helpers.js
- index.js
- mutations.js
```

Reply:
Maybe just link to Working with Vuex and extend this docs?